cart/urls.py

```diff
@@ -4,5 +4,6 @@
 app_name = 'cart'
 
 urlpatterns = [
-
+    url(r'^$', views.cart_detail, name='cart_detail'),
+    url(r'^add/(?P<book_id>\d+)/$', views.cart_add, name='cart_add'),
```

Comment:
Since you are using django 2+ why don't use [`path`](https://docs.djangoproject.com/en/2.1/ref/urls/#path)

Reply:
okey i will change it to path